From dffc7b6904d4f04f8d279f3b6d2c374b06ed1791 Mon Sep 17 00:00:00 2001
From: Stephen Barber <smbarber@chromium.org>
Date: Tue, 12 Feb 2019 17:15:33 -0800
Subject: [PATCH] client: fix goroutine leak in ExecContainer

The stdin channel from WebsocketSendStream needs to be consumed,
otherwise that goroutine will leak.

Signed-off-by: Stephen Barber <smbarber@chromium.org>
---
 client/lxd_containers.go | 1 +
 1 file changed, 1 insertion(+)

diff --git a/client/lxd_containers.go b/client/lxd_containers.go
index cd90913c..3aa9d73b 100644
--- a/client/lxd_containers.go
+++ b/client/lxd_containers.go
@@ -753,6 +753,7 @@ func (r *ProtocolLXD) ExecContainer(containerName string, exec api.ContainerExec
 
 				if fds["0"] != "" {
 					args.Stdin.Close()
+					<-dones[0]
 				}
 
 				for _, conn := range conns {
-- 
2.20.1.791.gb4d0f1c61a-goog

